messages/authen.go

```diff
@@ -70,6 +72,15 @@ func writeAuthenBytes(buf io.Writer, m Message) {
 		_ = binary.Write(buf, binary.BigEndian, prep.UI().Counter)
 	case ReqViewChange:
 		_ = binary.Write(buf, binary.BigEndian, m.NewView())
+	case ViewChange:
+		_ = binary.Write(buf, binary.BigEndian, m.NewView())
+		// There is no need to authenticate any other message
+		// content. Messages in the message log cannot be
+		// omitted/equivocated without it being detected,
+		// whereas any valid view-change certificate is
+		// equivalent to any other valid one. Moreover,
+		// messages in the message log and view-change
+		// certificate are authenticated on their own.
```

Comment:
Maybe I miss something, but It seems to me unclear why we don't have to include message log into signed data. I think that each message has a signature, but the order of messages in the log is not signed, so could a malicious replica intentionally reorder the log and disturb the replay after view change?

Reply:
Messages in the message log are required to be sequential in the counter value assigned by USIG. So any reordering, omission, or duplication of messages in the log will be detected when validating integrity of the ViewChange message.

I will add this explanation to the comment.